marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java

```diff
@@ -520,4 +532,12 @@ public void removeFieldFromAllProductDocuments(String fieldName) {
     Update update = new Update().unset(fieldName);
     mongoTemplate.updateMulti(new Query(), update, Product.class);
   }
+
+  public void transferComputedDataFromDB(Product product) {
+    Product persistedData = productRepository.findById(product.getId()).orElse(null);
+    if (Objects.isNull(persistedData)) {
```

Comment:
please use ifPresent to process data, avoid check null: e.g:
    productRepository.findById(product.getId()).ifPresent(persistedData -> {
      ProductFactory.transferComputedPersistedDataToProduct(persistedData,product);
    });